astrokat/noisediode.py

```diff
@@ -9,257 +9,419 @@
     from katcorelib import user_logger
 except ImportError:
     from .simulate import user_logger
+from . import _DEFAULT_LEAD_TIME
+from . import max_cycle_len
 
 
-_DEFAULT_LEAD_TIME = 5.0  # lead time [sec]
+def _get_max_cycle_len(kat):
+    if not kat.dry_run:
+        return max_cycle_len(kat.sensor.sub_band.get_value())
+    else:
+        return max_cycle_len('l')
+
+
+def _eval_lead_time_(lead_time):
+    # if not given, apply default value
+    lead_time = float(lead_time or _DEFAULT_LEAD_TIME)
```

Comment:
This has been added for an old `None` evaluation -- so the intension was to set default for both zero and `None`.

But looking at the code, it seems that this evaluation has become redundant in the restructuring -- I need to look at the code a little more

Reply:
Okay yes it is necessary -- because of the special implementation to allow the user to set the lead time from the config file

This will lead to a `None` value for the `lead_time` into the `pattern` and `trigger` functions, which is then updated to default

So the functionality is as expected

Reply:
Missed an obvious improvement -- with the `_DEFAULT_LEAD_TIME` moved to the `init` function this should be used at all times by all scripts

Which simplifies the implementation code very nicely and makes the `eval` function redundant
`_eval_lead_time_` has been removed